Sources/Sentry/SentryClient.m

```diff
@@ -56,12 +57,8 @@
 SentryClient ()
 
 @property (nonatomic, strong) SentryTransportAdapter *transportAdapter;
-@property (nonatomic, strong) SentryFileManager *fileManager;
```

Comment:
We don't have to redeclare these if SentryFileManager+TestProperties declares an interface _extension_ instead of a _category_. Did this in a few places:
- https://github.com/getsentry/sentry-cocoa/pull/2424/files#diff-4c24808dc36cb23d390cdeac2a97bc692ccf28a6f3da9a0bbc7bdd9cb468966aL10-R7)
- https://github.com/getsentry/sentry-cocoa/pull/2424/files#diff-a7a538405eee78d01496b490829d1cb182023f53d3b4c28151e8a60b9ff5a7c9L20-R20
- https://github.com/getsentry/sentry-cocoa/pull/2424/files#diff-a7a538405eee78d01496b490829d1cb182023f53d3b4c28151e8a60b9ff5a7c9L20-R20

The difference is that a category should be used when you don't own the implementation, like if you want to add a method to NSArray. But when you own the implementation and are just using additional headers to selectively expose methods to different things, class extensions should be used. 

From https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ObjectiveC/Chapters/ocCategories.html (with emphasis):
> Class extensions are like anonymous categories, except that the methods they declare must be implemented in the main @implementation block for the corresponding class...It is also generally common for a class to have a **publicly declared API and to then have additional methods declared privately for use solely by the class or the framework within which the class resides. Class extensions** allow you to declare additional required methods for a class in locations other than within the primary class @interface block

Reply:
Thank you 🙏